aiida_raspa/workchains/base.py

```diff
@@ -45,18 +45,162 @@ def report_error_handled(self, calculation, action):
         """Report an action taken for a calculation that has failed.
         This should be called in a registered error handler if its condition is met and an action was taken.
         :param calculation: the failed calculation node
-        :param action: a string message with the action taken
-        """
+        :param action: a string message with the action taken"""
+
         arguments = [calculation.process_label, calculation.pk, calculation.exit_status, calculation.exit_message]
         self.report('{}<{}> failed with exit status {}: {}'.format(*arguments))
         self.report('Action taken: {}'.format(action))
 
-
-@register_error_handler(RaspaBaseWorkChain, 570)
-def _handle_timeout(self, calculation):
-    """Error handler that restarts calculation finished with TIMEOUT ExitCode."""
-    if calculation.exit_status == RaspaCalculation.spec().exit_codes.TIMEOUT.status:
+    @process_handler(priority=570, exit_codes=RaspaCalculation.exit_codes.TIMEOUT, enabled=True)
+    def handle_timeout(self, calculation):
+        """Error handler that restarts calculation finished with TIMEOUT ExitCode."""
         self.report_error_handled(calculation, "Timeout handler. Adding remote folder as input to use binary restart.")
         self.ctx.inputs.parent_folder = calculation.outputs.remote_folder
+        return ProcessHandlerReport(False)
+
+    @process_handler(priority=400, enabled=False)
+    def check_widom_convergence(self, calculation):
+        """Checks whether a Widom particle insertion is converged. The check is based on the
+        error bar of the Henry coefficient."""
+
+        conv_threshold = 0.1
+        additional_cycle = 0
```

Comment:
Here, it needs to change to accept `conv_threshold` and `additional_cycle` as arguments.

Reply:
We cannot, unfortunately. Handlers do not accept input parameters.

Reply:
Oh, I see. Then, I think the solution would be keeping the `conv_threshold` as it is and for the `additiona_cycle` I suggest to have it either 
* as the product of a constant value (let's say 0.5) by initial value set by the user
* or a fixed value.
I, myself, vote for the first option. 
We can later do a benchmark to investigate more and modify the the code that would reflect handling of majority of cases.